sympy/core/trace.py

```diff
@@ -167,7 +167,13 @@ def doit(self, **kwargs):
 
         """
         if hasattr(self.args[0], '_eval_trace'):
-            return self.args[0]._eval_trace(indices=self.args[1])
+            if self.args[1] == ():
```

Comment:
When is `args[1]` an empty list? What error does `_eval_trace` give if `indices==()`?

I think this would be cleaner as

```
if self.args[1]:
    return self.args[0]._eval_trace(indices=self.args[1])
try:
    return self.args[0]._eval_trace()
except NotImplementedError:
    return Tr(self.args[0].doit())
```